src/us/kbase/workspace/test/kbase/KBaseWorkspaceConfigTest.java

```diff
@@ -89,6 +122,138 @@ private void listenerConstructFail(
 		}
 	}
 	
+	@Test
+	public void constructFailNullMap() throws Exception {
+		try {
+			new KBaseWorkspaceConfig(null);
+			fail("expected exception");
+		} catch (Exception got) {
+			TestCommon.assertExceptionCorrect(got, new NullPointerException("config"));
+		}
+	}
+	
+	@Test
+	public void loadFromFileFail() throws Exception {
+		final Path fakefile = Paths.get("superfakefile_princesspeachismyirlgf");
+		failLoadFromFile(fakefile, "thing", new KBaseWorkspaceConfigException(String.format(
+				"Could not read from configuration file superfakefile_princesspeachismyirlgf: " +
+						"%s (No such file or directory)", fakefile.toAbsolutePath())));
+
+		final Path notsofakefile = TEMP_DIR.resolve("imnotfakeimnotfake");
+		Files.write(notsofakefile, list(""), StandardCharsets.UTF_8);
+		failLoadFromFile(notsofakefile, "thing", new KBaseWorkspaceConfigException(String.format(
+				"The configuration file %s/imnotfakeimnotfake has no section thing", TEMP_DIR)));
+		
+		final Path realfile = TEMP_DIR.resolve("keepinitreal");
+		Files.write(realfile, list("[Workspace]", "mongodb-host=foo"), StandardCharsets.UTF_8);
+		failLoadFromFile(realfile, null, new KBaseWorkspaceConfigException(String.format(
+				"The configuration file %s/keepinitreal has no section null", TEMP_DIR)));
+		failLoadFromFile(realfile, "Workspacey", new KBaseWorkspaceConfigException(String.format(
+				"The configuration file %s/keepinitreal has no section Workspacey", TEMP_DIR)));
+	}
+	
+	private void failLoadFromFile(
+			final Path file,
+			final String section,
+			final Exception expected) {
+		try {
+			new KBaseWorkspaceConfig(file, section);
+			fail("expected exception");
+		} catch (Exception got) {
+			TestCommon.assertExceptionCorrect(got, expected);
+		}
+	}
+	
+	// TODO NOW_NEXT_PR reorder the args to be more sensible
+	// good heavens, what what absolutely ghastly code. I may faint
+	// hey kids, don't try this at home
+	private void assertConfigCorrect(
+			final Map<String, String> cfg,
```

Comment:
These arguments are in the order of the original asserts in the tests (other than one case)